armi/reactor/blocks.py

```diff
@@ -325,6 +325,15 @@ def getSmearDensity(self, cold=True):
 
         return smearDensity
 
+    def autoCreateSpatialGrids(self):
+        """
+        Blocks do not always have a spatialGrid from Blueprints, but, some Blocks can have their
+        spatialGrids inferred based on the multiplicty of their components.
+        This would add the ability to create a spatialGrid for a Block and give its children
+        the corresponding spatialLocators if certain conditions are met.
```

Comment:
Follow [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) format for docstrings. So short, one-line description, followed by a longer, more complete description. We have also decided that this can raise a ValueError if the implementation cannot be applied to a given block; there should be a [raises](https://numpydoc.readthedocs.io/en/latest/format.html#raises) section explaining what that means, and reminding the implementer/reassuring the user that in such a case, the block will be left un-modified.